api/content-script/listeners/wallet-settings-updated.ts

```diff
@@ -14,14 +14,19 @@ export const watchWalletSettingsUpdated = () => {
         window.location.host,
       );
 
+      window.postMessage(
+        ApiUtils.createApiResponse("kas:host_connected", isHostConnected),
+        window.location.origin,
+      );
+
       const account = isHostConnected
         ? await ApiUtils.getCurrentAccount()
         : null;
 
       window.postMessage(
         ApiUtils.createApiResponse(
           "kas:account_changed",
-          account?.address ?? null,
+          account ? account.address : null,
```

Comment:
If a dApp is connected to `testnet` and the user switches the network to `mainnet` we are going to leak the mainnet address to the dApp.

No a big issue, but it might come back to bite us later, don't you think?

Reply:
The mainnet and testnet have different host connected set. It would check the host is connected for the network when user switch the network. It only gives the address to the service which is trusted by users.
https://github.com/forbole/kastle/blob/main/components/screens/browser-api/ConnectConfirm.tsx#L52

The reason why to send the address back is to simplify the sdk to listen the balance change and update the address without `getWalletAddress`:
https://github.com/forbole/kastle-sdk/pull/1/files#diff-a937943b8a1ce57034a676bde56742e16d380cc806b2101f127d60fe460aaef4R24